File: SDKs/javascript/src/discord-messages.ts
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

/** Single DM in a Discord conversation (mirrors Unity IVXDirectMessage). */
export interface IVXDirectMessage {
  messageId: string;
  authorId: string;
  authorName: string;
  content: string;
  timestamp: number;
  isDisclosure: boolean;
  hasAdditionalContent: boolean;
  additionalContentDescription?: string;
  moderationMetadata?: Record<string, string>;
}

/** DM conversation summary row (mirrors Unity IVXDMSummary). */
export interface IVXDMSummary {
  userId: string;
  displayName: string;
  lastMessageId: string;
  lastMessageTimestamp: number;
}

/**
 * Discord Social SDK — direct messages: send, edit, history, summaries, chat visibility, deep links.
 * Stub surface; integrate with the native Discord Social SDK.
 */
export class IVXDiscordMessages {
  readonly isShowingChat = false;

  async sendDM(
    recipientId: string,
    message: string
  ): Promise<{ messageId: string }> {
    throw new Error('Not implemented');
  }

  async editDM(
    recipientId: string,
    messageId: string,
    newContent: string
  ): Promise<void> {
    throw new Error('Not implemented');
  }

  async getDMHistory(
    recipientId: string,
    limit = 50
  ): Promise<IVXDirectMessage[]> {
    throw new Error('Not implemented');
  }

  async getDMSummaries(): Promise<IVXDMSummary[]> {
    throw new Error('Not implemented');
  }

  setShowingChat(_showing: boolean): void {
    throw new Error('Not implemented');
  }

  openMessageInDiscord(_messageId: string): void {
    throw new Error('Not implemented');
  }

  openDMSettingsInDiscord(): void {
    throw new Error('Not implemented');
  }
}

File: SDKs/flutter/lib/src/discord_messages.dart
// Copyright (c) 2026 Intelli-verse-X
// MIT License — see LICENSE in the project root.

/// Discord DM line (Unity [IVXDirectMessage]).
class IVXDirectMessage {
  final String messageId;
  final String authorId;
  final String content;
  final int timestamp;

  const IVXDirectMessage({
    required this.messageId,
    required this.authorId,
    required this.content,
    required this.timestamp,
  });
}

/// DM conversation summary (Unity [IVXDMSummary]).
class IVXDMSummary {
  final String userId;
  final String displayName;
  final String lastMessageId;
  final int lastMessageTimestamp;

  const IVXDMSummary({
    required this.userId,
    required this.displayName,
    required this.lastMessageId,
    required this.lastMessageTimestamp,
  });
}

/// Direct messages API — stub matching Unity [IVXDiscordMessages].
class IVXDiscordMessages {
  IVXDiscordMessages._();
  static final IVXDiscordMessages instance = IVXDiscordMessages._();

  bool get isShowingChat => false;

  Future<String> sendDM(String recipientId, String message) async {
    throw UnimplementedError('IVXDiscordMessages.sendDM');
  }

  Future<void> editDM(
    String recipientId,
    String messageId,
    String newContent,
  ) async {
    throw UnimplementedError('IVXDiscordMessages.editDM');
  }

  Future<List<IVXDirectMessage>> getDMHistory(
    String recipientId, {
    int limit = 50,
  }) async {
    throw UnimplementedError('IVXDiscordMessages.getDMHistory');
  }

  Future<List<IVXDMSummary>> getDMSummaries() async {
    throw UnimplementedError('IVXDiscordMessages.getDMSummaries');
  }

  void setShowingChat(bool showing) {
    throw UnimplementedError('IVXDiscordMessages.setShowingChat');
  }

  void openMessageInDiscord(String messageId) {
    throw UnimplementedError('IVXDiscordMessages.openMessageInDiscord');
  }

  void openDMSettingsInDiscord() {
    throw UnimplementedError('IVXDiscordMessages.openDMSettingsInDiscord');
  }
}
